src/Piral.Blazor.Core/Extensions.cs

```diff
@@ -22,50 +38,57 @@ static class Extensions
                 allowedArgs.Add(type, allowed);
             }
 
-            return args
+            var allArgs = new List<IDictionary<string, object>> {args};
+            try
+            {
+                var routeParams = JsonSerializer.Deserialize<Match>(JsonSerializer.Serialize(args["match"]))?.@params;
+                if(!routeParams.IsNullOrEmpty()) allArgs.Add(routeParams);
+            }
+            catch(KeyNotFoundException) { }
+
+            var adjustedArgs = allArgs
+                .SelectMany(dict => dict)
                 .Where(m => allowed.Contains(m.Key))
                 .ToDictionary(m => m.Key, m => type.NormalizeValue(m.Key, m.Value));
+            
+            return adjustedArgs;
         }
-
+        
         public static object NormalizeValue(this Type type, string key, object value)
         {
             var property = type.GetProperty(key);

```

Comment:
I am not sure about the changes in this function. Why should it change? Why so many changes?

Reply:
You are right. This was due to the previous implementation where I had args passed in from js. Then, they would arrive as strings and needed to convert them to bools (bool.Parse...). Will revert!

Reply:
I just noticed that this is also necessary for the route params to function properly. If the parameter in Blazor needs to be an integer, the route param that is found in `args.match.params` will still be a string. So it needs to be parsed correctly.

Reply:
Alright, but let's split it up a bit to have the dedicated conversion in this function and have a target-specified conversion in another function. The target-specified conversion needs a string to start with (it only supports string->X where X may be one of Int32, Int64, Double, Float, Guid, ...).

Reply:
Solving this with `TypeDescriptor.GetConverter` as discussed